### DISCUSSION TITLE
请教下  quard_star 有硬件手册之类的吗？小白想知道对于一些地址、寄存器配置是这么来的

### DISCUSSION THREAD

**Discussion by later-3** [QQxiaoming/quard_star_tutorial#9] [General]

**Comment by later-3** [comment]
比如说 [QUARD_STAR_UART0]       = { 0x10000000,    0x1000 },  这里uart0 为啥是0x10000000的地址

**Comment by QQxiaoming** [comment]
非常抱歉，目前还没有比较完整归纳的文档，但这已经在我的计划中了。

就目前而言，也许可以通过阅读源码了解你想要的，整个系统的map可以在qemu-7.0.0/hw/riscv/quard_star.c:quard_star_memmap查看
具体的寄存器由于目前quard_star复用了大量qemu原本已有的ip模拟实现，因此你可以搜索对应的文档或直接阅读源码。quard_star.c目前已整理的比较容易阅读。

**Comment by QQxiaoming** [comment]
> 比如说 [QUARD_STAR_UART0] = { 0x10000000, 0x1000 }, 这里uart0 为啥是0x10000000的地址

对于这个地址的选择并没有什么特殊含义，在RISCV架构中似乎没有任何限制（但实际真实芯片架构系统总线实现可能会有限制），我只是把他放在这个地址了，你当然也可以改到其他地址。

**Comment by later-3** [comment]
好滴。感谢解答！
